packages/nx/src/native/hasher.rs

```diff
@@ -8,8 +8,13 @@ pub fn hash(content: &[u8]) -> String {
 }
 
 #[napi]
-pub fn hash_array(input: Vec<String>) -> String {
-    let joined = input.join(",");
+pub fn hash_array(input: Vec<Option<String>>) -> String {
+    let joined = input.iter().filter_map(|s| {
+        if s.is_none() {
+            trace!("Encountered None value in hash_array input: {:?}", input);
+        }
+        s.as_ref()
+    }).map(|s| s.as_str()).collect::<Vec<_>>().join(",");
```

Comment:
Use deref here so that we don't have to iterate through the vec again with the second map. 
```suggestion
    let joined = input.iter().filter_map(|s| {
        if s.is_none() {
            trace!("Encountered None value in hash_array input: {:?}", input);
        }
        s.as_deref()
    }).collect::<Vec<_>>().join(",");
```

If we didnt want to add the trace when with `is_none`, then we can do this, which is quicker:
```
input.into_iter()
        .flatten() // removes None and unwraps Some(String)
        .collect::<Vec<_>>()
        .join(",")
```

The time difference between the filter_map and flatten is miniscule, but still fun to know :D 
```
iter_filter_map_deref   time:   [68.009 µs 68.214 µs 68.425 µs]
                        change: [-1.2825% -0.5797% +0.2147%] (p = 0.15 > 0.05)
                        No change in performance detected.

into_iter_flatten       time:   [63.492 µs 63.643 µs 63.815 µs]
                        change: [-1.0299% -0.3958% +0.3385%] (p = 0.26 > 0.05)
                        No change in performance detected.
```

Reply:
for reference, heres the bench for the initial implementation with as_ref + as_str
```
iter_filter_map_ref     time:   [75.861 µs 76.254 µs 76.705 µs]
                        change: [-0.9861% -0.0645% +0.8159%] (p = 0.89 > 0.05)
                        No change in performance detected.
```